### ISSUE TITLE
Pix crashes or is not functional [closed]

### ISSUE THREAD

**Issue by Curius16** [linuxmint/pix#147]
Pix 2.8.4
Mint 20.3 _ 64 bits - Cinnamon 5.2.7 - Kernel 5.4.0.96 - Laptop Entroware Athena

Hello,
I've been using Pix since its creation (and before that I was using its predecessor Gthumb) without any problem (and so under diverse versions of Mint Cinnamon).

Since a few days now:
- the software crashes incessantly and under diverse circumstances,
- regularly it's unable to find pictures in a folder though it can do it after having being relaunched,
- regularly it refuses to copy a picture, showing a popup window and a red banner at the bottom of the window, both of them saying: "Impossible to perform the operation. Too much files opened." (see screenshot capture attached).

In this last case, in one occurence I had on purpose opened the software using the terminal in order to get the error messages; it gave this:

```
philippe@Athena:~$ pix

(pix:11919): Gtk-WARNING **: 12:46:19.590: Failed to register client: GDBus.Error:org.gnome.SessionManager.AlreadyRegistered: Unable to register client

(pix:11919): GLib-GObject-WARNING **: 12:46:19.610: g_type_class_add_private() called multiple times for the same type

(pix:11919): GStreamer-CRITICAL **: 12:46:34.906: g
![Pix Bug Copie](https://user-images.githubusercontent.com/98497463/151264896-5041084a-6191-4439-bbd6-d2659a250595.png)
st_poll_get_read_gpollfd: assertion 'set != NULL' failed

(pix:11919): GStreamer-CRITICAL **: 12:46:34.906: gst_poll_get_read_gpollfd: assertion 'set != NULL' failed

(pix:11919): GStreamer-CRITICAL **: 12:46:34.927: gst_poll_get_read_gpollfd: assertion 'set != NULL' failed

(pix:11919): GStreamer-CRITICAL **: 12:46:34.928: gst_poll_get_read_gpollfd: assertion 'set != NULL' failed

(pix:11919): GStreamer-CRITICAL **: 12:46:34.928: gst_poll_get_read_gpollfd: assertion 'set != NULL' failed

(pix:11919): GStreamer-CRITICAL **: 12:46:34.928: gst_bus_timed_pop_filtered: assertion 'timeout == 0 || bus->priv->poll != NULL' failed
```

Would someone have an idea about what's going on here?

Thanks
![Pix Bug Copie](https://user-images.githubusercontent.com/98497463/151265430-c15d6be6-4b9b-497b-9b8f-2786048629a9.png)
.

**Comment by Curius16** [comment]
Here's the screeshot capture for "Impossible to load the location.
![Pix Bug Impossible de charger-Copie](https://user-images.githubusercontent.com/98497463/151265070-6472ac95-9f78-4006-95dd-980ade9d413c.png)
"

**Comment by Curius16** [comment]
I have erased .config/pix et .local/share/pix
then uninstall the software with: sudo apt-get autoremove --purge pix
rebooted
then reinstall with: sudo apt-get install pix

and everything is exactly the same (cf attached picture: note the squares in the column with the files tree…).
![Bug Pix - Après réinstallation](https://user-images.githubusercontent.com/98497463/151265167-16caf0af-c5d3-47bd-b121-e23ea9a53864.png)

**Comment by Angedestenebres** [comment]
Hey :cat: 

I guess this is maybe linked to files naming. I didn't get this issue at all.
Moreover, in all your screenshots, the folder is the same "Téléchargements". 

Do you have some file with special character?
On one of your screens, we can see a folder with a weird name (only with squares) that's why I guess it can be linked to files naming.

**Comment by Curius16** [comment]
Bonjour Angedesténèbres et merci pour la réponse.

Les exemples montrés sont tous dans le dossier Téléchargements avec des photos que j'y avais mises pour la démonstration mais le problème survenait dans n'importe quel de mes dossiers de photos.

Je parle à l'imparfait car, de la même façon que le problème était survenu, il a disparu !

J'avais installé Xnview pour pallier l'absence d'un Pix fonctionnel et un jour où je retentais ma chance tout marchait à nouveau et je ne sais pas pourquoi (une mise à jour de paquet ?).

Aujourd'hui Pix fonctionne à nouveau à merveille et j'en suis très heureux car, à mon goût, c'est de très loin le logiciel qui combine à la perfection d'une part la richesse d'options et d'autre part la simplicité et la clarté. De mon point de vue on ne peut pas faire mieux.

Encore merci de t'être intéressé au bug !

**Comment by Angedestenebres** [comment]
Hey :cat: 

Ah ben tant mieux si c'est retombé en marche tout seul !

Du coup, je pense que tu vas pouvoir clôturer ton ticket :)

**Comment by Curius16** [comment]
C'est fait, merci Angedesténèbres.

**Comment by Spielmops** [comment]
I have the same problem and I would like to know, whats said in the last postings. Perhaps it could solve the problem. But on my installations it is 100% not related to special characters in filenames.

My wife and I have compact-cameras from Canon, my cam with 12Mp, my wifes with 20Mp, the filename is with both IMG_####.JPG. With the 20Mp-Images Pix crashes often, with 12Mp never. The output on the command-line is nothing of worth.

Spielmops

**Comment by mtwebster** [comment]
@Spielmops what distribution (mint 21, ubuntu..?) and what version of pix?

If you're on mint, can you open the 'System Reports` app and see if there are any crash reports? If there are, click the 'local files' button and attach the `crash.tar.gz` file here (if it's not too big).

You said you have the same problems as the OP - do you get the 'too many files open' error in pix?

Can you run (while that error is showing):
```
lsof -p `pidof pix` | wc -l
```
and report the number it prints.

Thanks

**Comment by Curius16** [comment]
Hello Spielmops,

I have just translated in english the preceding message that was in french. The translation is appended below the french text.

Moreover, every two years I make a clean install of Mint, wiping the system partition and most of the home partition (save a few config files but not those of Pix). So I am now under Mint 21 and a brand new Pix.

But as I said in the answer to Angesdesténèbres that I've just translated, before reinstalling, under Mint 20.3, one day without doing anything special, Pix worked perfectly anew!

I hope I'll occur also for you!
Best regards.
